src/TaskRunner.php

```diff
@@ -1,19 +1,51 @@
 <?php namespace CodeIgniter\Tasks;
 
+use CodeIgniter\CLI\CLI;
+
+/**
+ * Class TaskRunner
+ *
+ * @package CodeIgniter\Tasks
+ */
 class TaskRunner
 {
-    /**
-     * @var Scheduler
-     */
-    protected $scheduler;
+	/**
+	 * @var Scheduler
+	 */
+	protected $scheduler;
+
+	public function __construct()
+	{
+		$this->scheduler = service('scheduler');
+	}
 
-    public function __construct()
-    {
-        $this->scheduler = service('scheduler');
-    }
+	/**
+	 * The main entry point to run tasks within the system.
+	 * Also handles collecting output and sending out
+	 * notifications as necessary.
+	 */
+	public function run()
+	{
+		$tasks = $this->scheduler->getTasks();
 
-    public function run()
-    {
+		if (! count($tasks))
+		{
+			return;
+		}
 
-    }
+		try
+		{
+			foreach ($tasks as $task)
+			{
+				if ($task->shouldRun())
+				{
+					$task->run();
+				}
+			}
+		}
+		catch (\Throwable $e)
+		{
+			log_message('error', $e->getMessage(), $e->getTrace());
+		}
```

Comment:
I think the `try...catch` block should be just around `$task->run()` so we don't accidentally hide errors in `shouldRun()`.
```suggestion
		foreach ($tasks as $task)
		{
			if ($task->shouldRun())
			{
				try
				{
					$task->run();
				}
				catch (\Throwable $e)
				{
					log_message('error', $e->getMessage(), $e->getTrace());
				}
			}
		}
```

Reply:
@lonnieezell  I would still like to see this change.